template.py

```diff
@@ -28,25 +28,50 @@
 #
 # For more information, please refer to <http://unlicense.org/>
 
+'''
+YouCompleteMe configuration file template to be used in 'YCM-Generator'.
+'''
+
 # Needed because ur"" syntax is no longer supported
 from __future__ import unicode_literals
 
+import glob
 import os
-import ycm_core
 import re
 import subprocess
+import ycm_core
+
 
+# ========================== Configuration Options ============================
+GUESS_BUILD_DIRECTORY = False  # Experimental
+GUESS_INCLUDE_PATH = True
+# ========================== Configuration Options ============================
+
+# =========================== Constant Definitions ============================
+SOURCE_EXTENSIONS = ('.C', '.cpp', '.cxx', '.cc', '.c', '.m', '.mm')
+HEADER_EXTENSIONS = ('.H', '.h', '.hxx', '.hpp', '.hh')
```

Comment:
I think it would make more sense to just store these in lower case and use case-insensitive comparison